bin/deployment/update-server-xml.py

```diff
@@ -217,105 +215,104 @@ def _build_node(self):
-            "TLS_RSA_WITH_AES_256_CBC_SHA",
-            "TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA",
-            "TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA",
-            "TLS_ECDH_RSA_WITH_3DES_EDE_CBC_SHA",
-            "TLS_ECDH_RSA_WITH_AES_128_CBC_SHA",
-            "TLS_ECDH_RSA_WITH_AES_256_CBC_SHA",
-            "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
-        ])
+        # <Connector port="8443" protocol="HTTP/1.1"
+        #     scheme="https"
+        #     secure="true"
+        #     SSLEnabled="true"
+        #     maxThreads="150">
+        #
+        #     <SSLHostConfig certificateVerification="optional"
+        #         protocols="+TLSv1,+TLSv1.1,+TLSv1.2"
+        #         sslProtocol="TLS"
+        #         ciphers="SSL_RSA_WITH_3DES_EDE_CBC_SHA,TLS_RSA_WITH_AES_256_CBC_SHA,TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA,TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDH_RSA_WITH_3DES_EDE_CBC_SHA,TLS_ECDH_RSA_WITH_AES_128_CBC_SHA,TLS_ECDH_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA">
+        #
+        #         <Certificate
+        #                certificateFile="/etc/candlepin/certs/candlepin-ca.crt"
+        #                certificateKeyFile="/etc/candlepin/certs/candlepin-ca.key"
+        #                type="RSA"
+        #         />
+        #     </SSLHostConfig>
+        # </Connector>
 
-        # Setup our node configuration
         connector = libxml2.newNode("Connector")
-        self._add_attributes(connector, [
-            ("port", self.port),
-            ("protocol", "HTTP/1.1"),
-            ("SSLEnabled", "true"),
-            ("maxThreads", "150"),
-            ("scheme", "https"),
-            ("secure", "true"),
-        ])
+        self._add_attributes(connector, self.attributes)
 
-        sslconfig = libxml2.newNode("SSLHostConfig")
-        self._add_attributes(sslconfig, [
+        ssl_host_config = libxml2.newNode("SSLHostConfig")
+        self._add_attributes(ssl_host_config, [
             ("certificateVerification", "optional"),
             # Note SSLv3 is not included, to avoid poodle
             # For the time being, TLSv1 needs to stay enabled in Satellite deployments to support
             # existing python-rhsm based clients (RHEL5).
             ("protocols", "+TLSv1,+TLSv1.1,+TLSv1.2"),
             ("sslProtocol", "TLS"),
-            ("truststoreFile", "conf/keystore"),
-            ("truststorePassword", "password"),
-            ("truststoreType", "JKS"),
-            ("ciphers", ciphers),
+            ("ciphers", ",".join(SSL_CIPHERS))
         ])
 
         certificate = libxml2.newNode("Certificate")
         self._add_attributes(certificate, [
-            ("certificateKeystoreFile", "conf/keystore"),
-            ("certificateKeystorePassword", "password"),
-            ("certificateKeystoreType", "JKS"),
+            ('certificateFile', '/etc/candlepin/certs/candlepin-ca.crt'),
+            ('certificateKeyFile', '/etc/candlepin/certs/candlepin-ca.key'),
```

Comment:
Is there an advantage to using the direct PEM file? I was thinking using the keystore/truststore format would be more future-proof if we eventually wanted to have multiple CA certs (for splitting responsibilities on signing Identity and SCA certs with different keys). Although if that's hard to do it can always be left for later, since it is not an immediate need.

Reply:
We need the cert/key file anyway for CP (as it is expecting it to not be in a keystore). We could certainly change it, but it would require a couple things here:

- CP is updated to support both reading files directly for backward compatibility purposes, and reading them from a keystore (and all of the fallout that comes of that)
- Even more branching on the configuration surrounding FIPS mode, as *which* keystore format we use will depend on what is available via keytool and the security providers invoked by Tomcat's protocol/ssl handler.

It's important to note that this configuration for TC is not used outside of our team. Sat has their own setup process with cert generation and TC configuration, as does IT. If we're going to move forward on not having the PEM files on the filesystem, I would argue it's a separate task entirely as it has major impacts on our stakeholders and how CP itself functions.

Reply:
> We need the cert/key file anyway for CP (as it is expecting it to not be in a keystore). We could certainly change it, but it would require a couple things here:
> 
>     * CP is updated to support both reading files directly for backward compatibility purposes, and reading them from a keystore (and all of the fallout that comes of that)

Why would CP itself need to be updated for reading from keystore? that's Tomcat's job to handle. That was what it was always doing (the server.xm configured to use `keystoreFile`, etc.). In fact, using the cert/key PEM files directly (instead of using a keystore) was only recently introduced in [Tomcat 9](https://bz.apache.org/bugzilla/show_bug.cgi?id=59344).

> It's important to note that this configuration for TC is not used outside of our team. Sat has their own setup process with cert generation and TC configuration, as does IT. If we're going to move forward on not having the PEM files on the filesystem, I would argue it's a separate task entirely as it has major impacts on our stakeholders and how CP itself functions.

I wouldn't say not having PEM files on the filesystem at all. We'd still need to generate them and add them to the keystore, and tomcat would be reading the keystore only (as it always did before this PR). (and from what I understand, that's also what Satellite had been [doing](https://github.com/theforeman/puppet-candlepin/blob/49fec4a9624e2d65a87f6db919bb59f1fb17aa0c/templates/tomcat/server.xml.erb#L83-L87)).


>     * Even more branching on the configuration surrounding FIPS mode, as _which_ keystore format we use will depend on what is available via keytool and the security providers invoked by Tomcat's protocol/ssl handler.

That is definitely a pain, and probably why we should leave this as is, and use a keystore when we have a need for it in the future.

Reply:
> Why would CP itself need to be updated for reading from keystore? that's Tomcat's job to handle. That was what it was always doing (the server.xm configured to use keystoreFile, etc.). In fact, using the cert/key PEM files directly (instead of using a keystore) was only recently introduced in [Tomcat 9](https://bz.apache.org/bugzilla/show_bug.cgi?id=59344).

I'm aware, but CP itself also uses the certs/keys for setting the issuer and key identifier when creating certs. These are loaded directly from the PEM files (ConfigProperties.CA_CERT and ConfigProperties.CA_KEY), so no matter what we do with TC, CP still needs them available unless we're making a decision to move CP to load them cert/key from a keystore -- which invokes the need for backward compatibility I mentioned above, as well as determining which keystore formats we can use depending on system limitations.

Reply:
Looked into the [docs](https://tomcat.apache.org/tomcat-9.0-doc/config/http.html#Nested_Components) a bit more and apparently there is support  for multiple certificates to be associated with a single SSLHostConfig, so we don't need to use a keystore to have more than one cert.